### ISSUE TITLE
Union bug - "Some graphs are not named" despite named graphs

### ISSUE THREAD

**Issue by highly0** [igraph/python-igraph#629]
**Describe the bug**
The bug "some graphs are not named" appears despite the graphs being named (vertices have attributes called 'name')

**To reproduce**
```python
from igraph import *

# g1: 2 vertices and 1 edges
g1 = Graph(directed=True)
g1.add_vertices(2)
g1.vs["name"] = ["A", "B"]
g1.add_edges([(0, 1)])
print(g1.is_named()) # prints true

# g2: 3 vertices and 4 edges
g2 = Graph(directed=True)
g2.add_vertices(3)
g2.vs["name"] = ["B", "C", "D"]
g2.add_edges([(0, 1), (0, 2), (1, 2), (1, 0)])
print(g2.is_named()) # prints true

gs = [g1, g2]
g3 = Graph(directed=True)

# error here
g3 = g3.union(gs, byname=True)
```
error:
```
Traceback (most recent call last):
  File "/workspace/kbqa/test_path.py", line 50, in <module>
    graph = subgraph_retrieiver.get_subgraph(entities=entities, candidate=candidate)
  File "/workspace/kbqa/kbqa/wikidata/wikidata_subgraphs_retriever_igraph.py", line 83, in get_subgraph
    graph = self.subgraph_without_connection(paths)
  File "/workspace/kbqa/kbqa/wikidata/wikidata_subgraphs_retriever_igraph.py", line 134, in subgraph_without_connection
    res = res.union(graphs, byname=True)
  File "/opt/conda/lib/python3.10/site-packages/igraph/operators/methods.py", line 49, in _union
    return union([graph] + other, byname=byname)
  File "/opt/conda/lib/python3.10/site-packages/igraph/operators/functions.py", line 153, in union
    raise RuntimeError("Some graphs are not named")
RuntimeError: Some graphs are not named
```

I already took a look at the source code where this RunTimeError was raised. The following code was in the source code (where I had no issue with on my local)
```python
n_named = sum(g.is_named() for g in gs)
ngr = len(gs)
print(n_named != ngr) # prints False
```

**Version information**
0.10.3

**Comment by ntamas** [comment]
Hmmm, you are correct; it's an edge case but indeed it could be handled better.

`g3.union()` is an operation that merges `g1` and `g2` _into_ `g3`, so `g3` must also be named. Here `g3` has no vertices, so we should probably get away with _not_ requiring `g3` to be named because it does not matter in the end anyway. But I guess what you really want to do is:

```
from igraph import union

[...]

g3 = union([g1, g2], byname=True)
```

and then there is no need to create an empty graph just for the sake of having something that you could call `union()` on. Or you could also do `g3 = g1.union([g2], byname=True)`.

**Comment by ntamas** [comment]
After some consideration, I decided _not_ to act on this and consider this as the expected behaviour. `g3.union([g1, g2], byname=True)` is essentially the same as `union[g1, g2, g3], byname=True)` and the documentation states that all graphs must be named. `g3` is not named as it does not have a `"name"` attribute. I know that we could _guess_ the intention that the user wants to treat `g3` as if it were named, but in the spirit of the Zen of Python, I'd rather resist the temptation to guess.

In the end I decided to improve the error message a bit in 0d8c64a ; it now returns how many graphs were named or unnamed.
